app.js

```diff
@@ -1,4 +1,36 @@
 //  in this file create an express application - use the middle-ware built into express
 //  to serve up static files from the public directory (index.html and client.js - you 
 //  can also serve up css files from the public directory this way if you want)
-//  you need to support a '/trucks' endpoint, and a dynamic route for '/trucks:name'
\ No newline at end of file
+//  you need to support a '/trucks' endpoint, and a dynamic route for '/trucks:name'
+'use strict';
+var express = require('express');
+var app = express();
+var trucks = require('./trucks');
+
+app.use(express.static('public'));
+
+app.get('/trucks', function (request, response) {
+  response.json(trucks.getTrucks());
+});
+// the 'name' parameter will be available on the request.params object
+app.get('/trucks/:name', function (request, response) {
+  var truck = request.params.name;
+  response.json(trucks.getTruck(truck));
```

Comment:
Check here if getTruck returns something before sending it to the response and send a 404 status if it doesn't.